go/store/nbs/journal_chunk_source.go

```diff
@@ -209,6 +209,30 @@ func (s journalChunkSource) close() error {
 	return nil
 }
 
+func (s journalChunkSource) getAllChunkHashes(_ context.Context, out chan<- hash.Hash, wg *sync.WaitGroup) int {
+	chunkCount := s.journal.recordCount()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		s.journal.ranges.novel.Iter(func(k hash.Hash, v Range) (stop bool) {
+			out <- k
+			return false
+		})
+
+		s.journal.ranges.cached.Iter(func(k addr16, v Range) (stop bool) {
+			// Currently we only have 16bytes of the hash. The value returned here will have 4 0xFF bytes at the end.
+			var h hash.Hash
+			copy(h[:], k[:])
+			out <- h
+
+			return false
+		})
+	}()
```

Comment:
I don't think this is thread-safe. `novel` and `cached` are mutable and can't be used here without locking, right?

Reply:
fsck is an offline process. Shouldn't be anything editing the journal at this time. But I can stipulate that in a comment.